java/integTest/src/test/java/glide/cluster/CommandTests.java

```diff
@@ -556,4 +558,46 @@ public void time_with_route() {
                 "Time() result (" + serverTime[0] + ") should be greater than now (" + now + ")");
         assertTrue(Long.parseLong((String) serverTime[1]) < 1000000);
     }
+
+    @Test
+    @SneakyThrows
+    public void bgsave() {
+        var error = "Background save already in progress";
+        // bgsave may fail with given error and may keep failing on retries regardless of the
+        // timings/delays
+
+        var response = tryCommandWithExpectedError(() -> clusterClient.bgsave(false), error);
+        assertTrue(response.getValue() != null || response.getKey().startsWith("Background saving"));
+        Thread.sleep(2000); // next save call without delay will likely throw an error
```

Comment:
Why not remove the sleeps? Doesn't your tryCommandWithExpectedError already account for that?

Reply:
without sleep the chance of error is about ~100%. I wanted to have successful calls too sometimes.

Reply:
sometimes?  don't introduce more flakey commands.
If you have questions on to test something, ask on Slack - maybe Amazon can provide some hints.

Reply:
flakyness suppressed by `tryCommandWithExpectedError`, but with `sleep` the command succeeds much more othen

Reply:
I don't have strong opinions. I understand why you have added the sleeps but at the end of the day you have added this function tryCommandWithExpectedError (it quite doesn't matter if it succeeds or fails at this point). I would ask the AWS team for their input or make this function much simpler so it just succeeds.